lib/wibox/init.lua

```diff
@@ -36,6 +37,272 @@ function wibox:set_widget(widget)
     self._drawable:set_widget(widget)
 end
 
+--- Set a declarative widget hierarchy description.
```

Comment:
Yes, I think I should make a `.md` like the user contributor guide. I could also move `awful.rule` to one too. They are very high level, very general features users are much more likely to use compared to a deep API doc page. What do you think (and what do @blueyed think)?